controllers/storagecluster/reconcile.go

```diff
@@ -159,14 +161,24 @@ func (r *StorageClusterReconciler) Reconcile(ctx context.Context, request reconc
 		return reconcile.Result{}, err
 	}
 
+	var err error
+	availableCrds, err := statusutil.MapCRDAvailability(ctx, r.Client, statusutil.GetCrds()...)
+	if err != nil {
+		r.Log.Error(err, "error fetching the CRDs")
+		return reconcile.Result{}, err
+	}
+	if !reflect.DeepEqual(availableCrds, r.AvailableCrds) {
+		r.Log.Info("CustomResourceDefinitions created/deleted. Restarting process.")
+		os.Exit(util.RestartExitCode)
+	}
```

Comment:
Why are we trying to be generic here? just check the CRDs you are concerned about

Reply:
There can be multiple CRDs we can be concerned about, for this case, we are concerned about ClusterClaim CRD and Virtualization CRD (used for creation of the virt-rbd storageClass). In case the CRDs get created/deleted we need to exit from the storage cluster controller, hence we can be generic here and exit if any one of the values is different from what we calculated in the main